### PR TITLE
Add task and status objects as well as the task repository

### PR DESCRIPTION
**Pull request by skylarnam**
## What?
Add task and status objects as well as the task repository

## Why?
We want to start adding domain objects for the task and status, as well as the task repository to interact with the database.

## How?
Add a few base files and set up a task repository to interact with the database, ready to be called from the main function.

## Testing
- [x] Ensured the build passes

### CODE REVIEW

File: crates/app/src/main.rs [hunk 1]
```diff
@@ -54,5 +55,8 @@ async fn main() -> anyhow::Result<()> {
         .await
         .unwrap();
 
+    let task_repo = DomainTaskRepository::new(pool.clone());
+    // let tasks = task_repo.get_tasks().await?;
+
```
Comment (fm100) [inline crates/app/src/main.rs]: We may want to move this into `service_registry.rs`.

File: crates/domain/src/repositories/task_repository.rs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                    created_at: row.get("created_at"),
+                    updated_at: row.get("updated_at"),
+                })
+                .fetch_all(&self.pool)
+                .await?;
+        Ok(tasks)
+    }
+
+    async fn create_task(&self, task: Task) -> anyhow::Result<Task> {
+        let now = Utc::now().naive_utc();
```
Comment (fm100) [inline crates/domain/src/repositories/task_repository.rs]: Why do we want to remove timezone information?

File: crates/app/src/main.rs (hunk 1 above)
Comment (skylarnam) [reply crates/app/src/main.rs]: okie

File: crates/domain/src/repositories/task_repository.rs (hunk 2 above)
Comment (skylarnam) [reply crates/domain/src/repositories/task_repository.rs]: Seems like `Utc::now().naive_utc()` is commonly used when you need a datetime for purposes like storing in a database, logging, or comparing without explicitly requiring timezone handling. (So the timezone is always recorded in the same way).

Comment (fm100) [reply crates/domain/src/repositories/task_repository.rs]: https://docs.rs/chrono/latest/chrono/naive/struct.NaiveDateTime.html

Yes, but we may want to keep the timezone information and the column time is also a `TIMESTAMPTZ`, which is with timezone information already.

File: crates/domain/src/models/task.rs [hunk 3]
```diff
@@ -11,8 +11,8 @@ pub struct Task {
     pub description: Option<String>,
     pub status: Status,
     pub tags: Option<Vec<String>>,
-    pub created_at: NaiveDateTime,
-    pub updated_at: NaiveDateTime,
+    pub created_at: DateTime<Utc>,
```
Comment (fm100) [inline crates/domain/src/models/task.rs:15]: Let's add `#[serde(rename = "createdAt")]` and `#[serde(rename = "updatedAt")]` to make the json object key camelCase. Same thing applies to line 9.

**Review by fm100 — APPROVED** [review]
LGTM, with one nit.

File: crates/domain/src/models/task.rs (hunk 3 above)
Comment (skylarnam) [reply crates/domain/src/models/task.rs:15]: Can I just do `#[serde(rename_all = "camelCase")]`?

Comment (fm100) [reply crates/domain/src/models/task.rs:15]: Ah, I didn't know that is possible. That sounds simpler.

**Review by Picbridge — APPROVED** [review]

**Pull request merged**